core/dbt/parser/schemas.py

```diff
@@ -881,15 +884,31 @@ def parse_patch(self, block: TargetBlock[NodeTarget], refs: ParserRef) -> None:
                 f"Unexpected yaml_key {patch.yaml_key} for patch in "
                 f"file {source_file.path.original_file_path}"
             )
+        # handle disabled nodes
         if unique_id is None:
             # Node might be disabled. Following call returns list of matching disabled nodes
             found_nodes = self.manifest.disabled_lookup.find(patch.name, patch.package_name)
             if found_nodes:
-                # There might be multiple disabled nodes for this model
+                if len(found_nodes) > 1 and patch.config.get("enabled"):
+                    # There are multiple disabled nodes for this model and the schema file wants to enable one.
+                    # We have no way to know which one to enable.
+                    msg = (
+                        f"Found {len(found_nodes)} matching disabled nodes for '{patch.name}'. "
+                        "Multiple nodes for the same unique id cannot be disabled in the schema "
+                        "file. They must be disabled in `dbt_project.yml` or in the sql files."
+                    )
+                    raise ParsingException(msg)
+
+                # all nodes in the disabled dict have the same unique_id so just grab the first one
+                # to append with the uniqe id
+                source_file.append_patch(patch.yaml_key, found_nodes[0].unique_id)
                 for node in found_nodes:
-                    # We're saving the patch_path because we need to schedule
-                    # re-application of the patch in partial parsing.
                     node.patch_path = source_file.file_id
+                    # re-calculate the node config with the patch config.  Always do this
+                    # for the case when no config is set to ensure the default of true gets captured
+                    self.patch_node_config(node, patch)
```

Comment:
I think we can do "if patch.config:" before doing patch_node_config. If there's no config in this patch, then there's nothing to change.